File: app/src/main/java/net/nashihara/naroureader/presenter/DownloadedRecyclerPresenter.java
package net.nashihara.naroureader.presenter;

import android.support.v7.widget.RecyclerView;

import net.nashihara.naroureader.entities.Novel4Realm;
import net.nashihara.naroureader.utils.RealmUtils;
import net.nashihara.naroureader.views.DownloadedRecyclerView;

import java.util.ArrayList;

import io.realm.Realm;
import io.realm.RealmResults;

public class DownloadedRecyclerPresenter implements Presenter<DownloadedRecyclerView> {

    private DownloadedRecyclerView view;

    public DownloadedRecyclerPresenter(DownloadedRecyclerView view) {
        attach(view);
    }

    @Override
    public void attach(DownloadedRecyclerView view) {
        this.view = view;
    }

    @Override
    public void detach() {
        view = null;
    }

    public void fetchDownloadedNovels() {
        Realm realm = RealmUtils.getRealm(view.getContext());
        RealmResults<Novel4Realm> results = realm.where(Novel4Realm.class).equalTo("isDownload", true).findAll();

        ArrayList<Novel4Realm> novels = new ArrayList<>();
        for (Novel4Realm novel4Realm : results) {
            novels.add(novel4Realm);
        }

        view.showDownloadedNovels(novels);
    }
}

File: app/src/main/java/net/nashihara/naroureader/presenter/DownloadedRecyclerPresenter.kt
package net.nashihara.naroureader.presenter

import android.content.Context
import android.support.v7.widget.RecyclerView

import net.nashihara.naroureader.entities.Novel4Realm
import net.nashihara.naroureader.utils.RealmUtils
import net.nashihara.naroureader.views.DownloadedRecyclerView

import java.util.ArrayList

import io.realm.Realm
import io.realm.RealmResults

class DownloadedRecyclerPresenter(view: DownloadedRecyclerView) : Presenter<DownloadedRecyclerView> {

    private var view: DownloadedRecyclerView? = null

    init {
        attach(view)
    }

    override fun attach(view: DownloadedRecyclerView) {
        this.view = view
    }

    override fun detach() {
        view = null
    }

    fun fetchDownloadedNovels(context: Context) {
        val realm = RealmUtils.getRealm()
        val results = realm.where(Novel4Realm::class.java).equalTo("isDownload", true).findAll()
        view?.showDownloadedNovels(results.toList())
    }
}
